### DISCUSSION TITLE
Dax Studio open PowerBI file without running Power BI Desktop and automated export to SQL process

### DISCUSSION THREAD

**Discussion by ShockMonster** [DaxStudio/DaxStudio#1056] [Q&A] [answered]
Criteria: Access a PowerBI file to export its table contents to Azure SQL Server tables once per day on a daily schedule.

Is there is any way to not require a person to manually open up Desktop PowerBI and do the export? 

We want to be able to automate it to happen once daily. Can Dax Studio support an automated process to access a PowerBI file and export the contents?

**Comment by dgosbell** [comment, marked as answer]
It's not possible to connect to a pbix file without desktop being open. You would have to automate the opening of Desktop. But DAX Studio does not currently support automated exports. It's an item on my list, but I don't have any timeframe for this yet.

**Reply by ahsenmajid** [reply]
@dgosbell, given that Dax Studio now has it's own command line utility, would it be feasible to write a PowerShell script to open each pbix in a folder and run DSCMD commands to export the data?

**Reply by dgosbell** [reply]
I was thinking about this recently. Probably the one missing piece is the ability for the command line tool to either open or find an already opened pbix file by name. The trick is that we need to find the dynamic IP port that Desktop has assigned. I have this functionality in the connection dialog so I think I just need to add a parameter to the command line tool where you can either pass in the pbix filename or the full path.

**Reply by Rakshit-Sharma-001** [reply]
@dgosbell  
Is there any other way to connect to a BI file besides the current local .pbix report? Perhaps through a connection to the Power BI Service using an account or some alternative method that could bypass the limitation of having to open the file in Power BI Desktop?

**Reply by marcosqlbi** [reply]
Use the XMLA endpoint in the "Tabular Server" option of the Connect dialog box.

![image](https://github.com/user-attachments/assets/b5f391b5-de6b-4cf5-82b4-424a9f0324b9)

**Reply by Rakshit-Sharma-001** [reply]
Thanks for the help

**Reply by iveniveniveniven** [reply]
@dgosbell does the command line tool has the functionality to use an opened pbix file like you mentioned above? Thanks!

**Reply by dgosbell** [reply]
@iveniveniveniven - not yet, but it's part of the next release
